library/ssl_cli.c

```diff
@@ -642,7 +637,8 @@ static void ssl_write_alpn_ext( mbedtls_ssl_context *ssl,
     /* Skip writing extension and list length for now */
     p += 4;
 
-    for( cur = ssl->conf->alpn_list; *cur != NULL; cur++ )
+    for( cur = ssl->conf->alpn_list;
+	 p < buf + 6 + alpnlen && cur != NULL; cur++ )
```

Comment:
This lop condition looks a bit redundant as you have already computed the value alpnlen and checked that there is enough space in the buffer